sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonParser.scala

```diff
@@ -35,184 +34,337 @@ import org.apache.spark.util.Utils
 
 private[json] class SparkSQLJsonProcessingException(msg: String) extends RuntimeException(msg)
 
-object JacksonParser extends Logging {
+private[sql] class JacksonParser(
+    schema: StructType,
+    columnNameOfCorruptRecord: String,
+    options: JSONOptions) extends Logging {
 
-  def parse(
-      input: RDD[String],
-      schema: StructType,
-      columnNameOfCorruptRecords: String,
-      configOptions: JSONOptions): RDD[InternalRow] = {
+  import com.fasterxml.jackson.core.JsonToken._
 
-    input.mapPartitions { iter =>
-      parseJson(iter, schema, columnNameOfCorruptRecords, configOptions)
+  // A `ValueConverter` is responsible for converting a value from `JsonParser`
+  // to a value in a field for `InternalRow`.
+  private type ValueConverter = (JsonParser) => Any
+
+  // `ValueConverter`s for the root schema for all fields in the schema
+  private val rootConverter: ValueConverter = makeRootConverter(schema)
+
+  private val factory = new JsonFactory()
+  options.setJacksonOptions(factory)
+
+  private def failedConversion(
+      parser: JsonParser,
+      dataType: DataType): Any = parser.getCurrentToken match {
+    case _ if parser.getTextLength < 1 =>
```

Comment:
Sure! For `StringType`, it does not fail and call this function (https://github.com/apache/spark/pull/14102/files#diff-8affe5ec7d691943a88e43eb30af656eR231). So, for other types, when the actual data is empty, we should return `null` instead of empty string because it is guaranteed that it is not `StringType`.